clang/lib/CodeGen/Targets/SPIR.cpp

```diff
@@ -38,9 +38,20 @@ class CommonSPIRABIInfo : public DefaultABIInfo {
 ABIArgInfo CommonSPIRABIInfo::classifyKernelArgumentType(QualType Ty) const {
   Ty = useFirstFieldIfTransparentUnion(Ty);
 
-  if (getContext().getLangOpts().SYCLIsDevice && isAggregateTypeForABI(Ty)) {
+  if (getContext().getLangOpts().SYCLIsDevice) {
+    if (const BuiltinType *BT = Ty->getAs<BuiltinType>()) {
+      switch (BT->getKind()) {
+      case BuiltinType::Bool:
+        // Bool / i1 isn't a legal SPIR-V type.
```

Comment:
```suggestion
        // Bool / i1 isn't a legal kernel argument in SPIR-V.
```
I'm also getting confused by SPIR and SPIR-V. It is SYCL that generates SPIR-V via using SPIR target when generating LLVM IR, but AFAIK they are not the same. I wonder if sycl should simply start SPIRV target that now exists and we can avoid doing fixes like that.

Reply:
>  but AFAIK they are not the same

They aren't the same, yet so entangled it doesn't really make sense to separate them:

SPIR is coming from the SPIR 1.2 KHR specs which specify an LLVM 3.2 format as the kernel IR. It specifies metadata format, CC, allowed attributes and instructions and the address space mapping. Note that because it is based on LLVM 3.2, it is no longer possible to emit a valid SPIR module using clang tip (and hasn't been possible for years as the binary format changed).

When the SPIR-V translator and then backend were started, they used the SPIR convention as the input LLVM IR they would consume. So to produce a SPIR-V module you need to use the SPIR convention.